src/glados/ASR/ctc_asr.py

```
@@ -0,0 +1,266 @@
+            sess_options=session_opts,
+            providers=providers,
+        )
+
+        # 3. Load the vocabulary from the YAML configuration file
+        if "labels" not in self.config:
+            raise ValueError("YAML missing 'labels' section for vocabulary configuration.")
+        self.idx2token: dict[int, str] = dict(enumerate(self.config["labels"]))
+        num_tokens = len(self.idx2token)
+        if num_tokens != self.config["decoder"]["vocab_size"]:
+            raise ValueError(
+                f"Mismatch between number of tokens in vocabulary ({num_tokens}) and decoder vocab size ({self.config['decoder']['vocab_size']})."
```

Ruff (0.8.2)

Line too long (146 > 120)

(E501)